docs/agent_api.rst

```diff
@@ -3,4 +3,313 @@
+:op_result: The result status of this operation.
+
+------
+
+.. code-block:: cpp
+
+    bool create_participant_by_ref(uint32_t client_key, uint16_t participant_id, int16_t domain_id, const char* ref, uint8_t flag, OpResult& op_result);
+
+Creates a DDS participant for a certain client, given a reference to its description hosted in a certain XML descriptor file.
+
+The references file must have been previously loaded by the agent.
+
+The participant will act the entrypoint for the rest of DDS entities to be created.
+
+Returns `true` if the creation was successful, or `false` otherwise.
+
+:client_key: The identifier of the `ProxyClient` to which the resulting participant will be attached to.
+:participant_id: The identifier of the participant to be created.
+:domain_id: The DDS domain ID associated to the participant.
+:ref: The reference tag which will retrieve the participant description from the references file, previously loaded by the agent.
+:flag: It determines the creation mode of the new participant (see :ref:`creation_mode_table`).
+:op_result: The result status of this operation.
+
+..
+    TODO: Add reference to creation mode section (xml/ref).
+
+------
+
+.. code-block:: cpp
+
+    bool delete_participant(uint32_t client_key, uint16_t participant_id, OpResult& op_result);
+
+Removes a DDS participant from a certain client proxy.
+Returns `true` if the participant was deleted, or `false` otherwise.
+
+:client_key: The identifier of the `ProxyClient` from which the participant must be deleted.
+:participant_id: The ID of the participant to be deleted.
+:op_result: The result status of the operation.
+
+------
+
+.. code-block:: cpp
+
+    bool create_<entity>_by_xml(uint32_t client_key, uint16_t <entity>_id, uint16_t <associated_entity>_id, const char* xml, uint8_t flag, OpResult& op_result);
+
+Creates a certain DDS entity attached to an existing `ProxyClient`, given its client key.
+
+An XML must be provided, containing the DDS description of the entity to be created.
+
+There are as many methods available as existing DDS entities, replacing the parameters *<entity>* and *<associated_entity>* as follows:
+
+.. _existing_entities_and_associated_entities:
+
+*Agent's API available DDS entities and their associated entities*
+******************************************************************
+
+=============== =========================
+**<entity>**    **<associated_entity>**
+=============== =========================
+topic           participant
+publisher       participant
+subscriber      participant
+datawriter      publisher
+datareader      subscriber
+requester       participant
+replier         participant
+=============== =========================
+
+This operation returns `true` if the entity is successfully created and linked to its associated entity (which must previously exist in the given `ProxyClient`), or false otherwise.
```

Comment:
```suggestion
This operation returns ``true`` if the entity is successfully created and linked to its associated entity (which must previously exist in the given ``ProxyClient``), ``false`` otherwise.
```